exchange/rpc.go

```diff
@@ -229,6 +236,9 @@ func PutExchangeDevice(httpClientFactory *config.HTTPClientFactory, deviceId str
 			}
 		} else {
 			glog.V(3).Infof(rpclogString(fmt.Sprintf("put device %v to exchange %v", deviceId, pdr)))
+			if cachedNode != nil {
+				UpdateCacheNodePutWriteThru(GetOrg(deviceId), GetId(deviceId), cachedNode, pdr)
```

Comment:
ok, I see what you're doing here and I understand it. The problem with this approach will occur some time in the future when we add a new field to the node (pdr) and forget to update the UpdateCacheNodePutWriteThru function. Is there a test we could add to the unit tests or something we could add to the code that will catch this problem?

Reply:
Couldn't think of a way to do this with a unit test since the test would need to know about a new field in PutDeviceRequest to catch it. Instead, I added a check that all fields in the pdr are used to update the cache. If there is a field in the pdr not used to update the cache, an error will be logged and the cached device deleted

Reply:
Ok, that's a creative solution. Can you change the Infof that logs the message to an Errorf.

Reply:
Fixed